src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -328,6 +329,35 @@ private static bool IsWhite(char ch)
                 return null;
             }
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static unsafe bool AllowHyphenDuringParsing(NumberFormatInfo info)
+            {
+                string negativeSign = info.NegativeSign;
+                return negativeSign.Length == 1 &&
+                       negativeSign[0] switch {
+                           '\u2012' or         // Figure Dash
```

Comment:
I think I'm missing something, shouldn't this return true for regular hyphen \u002d? Even if it's not functionally necessary, it seems like a confusing method name if it doesn't return true for that.

Reply:
Or maybe AllowAlternativeHyphenDuringParsing might be a better name?

Reply:
No. If we have the negative sign as '-' then we don't need to do anything. The fix here is to allow parsing when using hyphen in the parsed string. when we have NegativeSign as hyphen it is already working today without any fix.

Reply:
> Or maybe AllowAlternativeHyphenDuringParsing might be a better name?

No, we are allowing parsing the hyphen and not the alternative hyphens. 

The fix here is when having a culture using alternative hyphens as negative sign, we'll allow parsing strings using the hyphen 0+002D `-`.

Reply:
Ah, you're quite right. I was thinking backwards.